### PR TITLE
🏗️ Build mixin + spiders: bisnd_mc

### PR DESCRIPTION
**Pull request by SimmonsRitchie**
## What's this PR do?
Adds a mixin and spiders to scrape multiple commission/boards from the website of the City of Mandan in Mandan, ND. The mixin that handles the core scraping logic is called `bisnd_mc`.

## Why are we doing this?
Requested by our site partners.

## Steps to manually test

1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spiders:
```
scrapy list | grep 'bisnd_mc' | xargs -I {} scrapy crawl {} -O {}.csv
```
4. Monitor the output and ensure no errors are raised.

5. Inspect the CSVs to ensure the data looks valid. You can view the value in the "source" column of the rows to ensure the data matches.

## Are there any smells or added technical debt to note?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206618632217643

### CODE REVIEW

File: city_scrapers/mixins/mc.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    base_url = "https://mandannd.api.civicclerk.com"
+    timezone = "America/Chicago"
+    name = None
+    agency = None
+    category_id = None
+
+    def start_requests(self):
+        """
+        sdfsdfsdf
```
Comment (LienDang) [inline city_scrapers/mixins/mc.py]: Dummy text?

File: city_scrapers/mixins/mc.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        """
+        # Calculate dates for one month prior and one year ahead
+        today = datetime.today()
+        one_month_prior = today - relativedelta(months=1)
+        one_year_ahead = today + relativedelta(months=6)
```
Comment (LienDang) [inline city_scrapers/mixins/mc.py]: Naming issue `one_year_ahead` vs. `half_year_ahead`?

File: city_scrapers/mixins/mc.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        """
+        Parse published files into links.
+        """
+        links = []
+        if "publishedFiles" in item and len(item["publishedFiles"]) > 0:
+            for file in item["publishedFiles"]:
+                links.append(
+                    {
+                        "title": file["name"],
+                        "href": f"https://mandannd.api.civicclerk.com/v1/Meetings/GetMeetingFileStream(fileId={file['fileId']},plainText=false)",  # noqa
```
Comment (LienDang) [inline city_scrapers/mixins/mc.py]: You can replace `https://mandannd.api.civicclerk.com` with `{self.base_url}`

File: city_scrapers/mixins/mc.py (hunk 1 above)
Comment (SimmonsRitchie) [reply city_scrapers/mixins/mc.py]: Yep. Now fixed!

File: city_scrapers/mixins/mc.py (hunk 2 above)
Comment (SimmonsRitchie) [reply city_scrapers/mixins/mc.py]: Yep. Now fixed!

File: city_scrapers/mixins/mc.py (hunk 3 above)
Comment (SimmonsRitchie) [reply city_scrapers/mixins/mc.py]: Yep. That's def tidier. Thanks, @LienDang.

**Review by LienDang — APPROVED** [review]
LGTM

**Pull request merged**